_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,334 @@
+
+**Warning**
+
+Setting up the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) as described in this blog post requires disabling HTTP basic authentication for the Cruise Control REST API.
+Without authentication, anybody with access to the Cruise Control API can use all the available endpoints and trigger administrative actions such as rebalancing partitions or decommissioning brokers.
+If using the Cruise Control UI with a Strimzi-managed Kafka cluster, it is best to stick with using it solely for monitoring your cluster, not for operating on it.
+That being said, use at your own risk!
+
+![Cruise Control Frontend](/assets/images/posts/2019-08-08-cruise-control-frontend.png)
+
+
+## Motivation
+
+We have been getting a lot of requests for how to set up the Cruise Control UI with a Strimzi-managed Kafka cluster.
+Maybe you want a pretty dashboard to view your cluster status.
+Maybe you want to compare optimization proposals without having to open a text editor.
+Maybe you are just a visual person.
+Regardless of the reason, we want to provide some basic instructions for getting you started!
+
+## Prerequisites
+
+For starters we need a Strimzi-managed Kafka cluster with Cruise Control enabled.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-1.png)
+
+Since Strimzi does not support creating custom Cruise Control API user roles we need to disable HTTP basic authentication settings.
+
+We can do so by configuring Cruise Control within a `Kafka` resource like this:
+
+``` 
+# kafka-cruise-control.yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  cruiseControl:
+    config:
+      webserver.security.enable: false
+```
+
+After configuring our Strimzi-managed Kafka cluster, we can deploy and connect the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) in one of two ways:
+
+* METHOD ONE: Customizing the Cruise Control pod
+* METHOD TWO: Creating a dedicated Cruise Control UI pod
+
+## METHOD ONE: Customizing the Cruise Control pod
+
+With our Strimzi-managed Kafka cluster deployed, we can create and deploy a custom Cruise Control image which includes the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) application inside of it.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-1.png)
+
+
+### Inside the custom Cruise Control container
+
+Here the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) will run alongside the Cruise Control application in the same container making its requests directly to the Cruise Control REST API.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-3.png)
+
+### Building the Cruise Control UI container
+
+We can base our custom Cruise Control image on the original Strimzi Kafka image and install the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) bits on top:
+
+```bash
+#Dockerfile
+FROM quay.io/strimzi/kafka:latest-kafka-3.2.3
```

Comment:
The latest image is constantly changing and basically does not produce a reproducible build. You should use the last release image. You should probably also clarify how it should be updated with every Strimzi upgrade.

Reply:
Updated the image tag and added a comment